### ISSUE TITLE
get "EvalError: Code generation from strings disallowed for this   context" when deploying to cloudflare worker

### ISSUE THREAD

**Issue by implicit-invocation** [elysiajs/elysia#58]
```
import { Elysia } from "elysia";
import { cors } from "@elysiajs/cors";
import { swagger } from "@elysiajs/swagger";

const app = new Elysia()
  .use(swagger())
  .use(cors())
  .get("/", () => "Hello Elysia");

export default {
  async fetch(req: Request) {
    return await app.handle(req);
  },
};
```

I tried to deploy this using wrangler, but it seems that the build contains some eval call.
```
  Uncaught Error: EvalError: Code generation from strings disallowed for this
  context
    at index.js:5549:15 in composeHandler
    at index.js:7173:407 in add
    at index.js:7290:21 in get
    at index.js:7533:8
    at index.js:7283:18 in use
    at index.js:7595:20
   [code: 10021]
```

Is there any way to work around this?

**Comment by implicit-invocation** [comment]
it seems that elysia actually uses code generation from string [https://github.com/elysiajs/elysia/blob/main/src/compose.ts](https://github.com/elysiajs/elysia/blob/main/src/compose.ts)

**Comment by ryoppippi** [comment]
I got the same issue
I hope this issue to be solved

**Comment by SaltyAom** [comment]
In an upcoming release of Elysia, Code Generation will be disabled by default on Cloudflare Worker, you can try out on Elysia 0.6.0-alpha.2 to see if it's solved the problem. 

b2df605

Please leave this issue open until 0.6.0 stable is released.

**Comment by ryoppippi** [comment]
I tried.
When I write code like this 
```ts
const app = new Elysia({
  aot: false,
})
  .get("/", () => "Hello Elysia")
export default app;
```
it works fine.
But if I write code like this

```ts
const app = new Elysia({
  aot: false,
})
  .get("/", () => "Hello Elysia")
  .post(
    "/post",
    async (event) => {
      const body = event.body;
      return body;
    },
    {
      body: t.Object({
        username: t.String(),
        password: t.String(),
      }),
    }
  );

export default app;
```
It causes errors like this 
```sh
service core:user:hi-elysia: Uncaught Error: EvalError: Code generation from strings disallowed for this context
  at index.js:3307:45 in Compile
  at index.js:7118:135 in getSchemaValidator
  at index.js:8289:22 in add
  at index.js:8413:17 in post
  at index.js:8503:35
```

**Comment by ryoppippi** [comment]
OK, I think this error comes from typebox 

```ts
// node_modules/.pnpm/@sinclair+typebox@0.29.6/node_modules/@sinclair/typebox/compiler/compiler.js

...
      function Compile(schema, references = []) {
        const code = Code(schema, references, { language: "javascript" });
        const customs = new Map(state.customs);
        const compiledFunction = globalThis.Function("custom", "format", "hash", code);
        const checkFunction = compiledFunction((kind, schema_key, value) => {
          if (!Types.TypeRegistry.Has(kind) || !customs.has(schema_key))
            return false;
          const schema2 = customs.get(schema_key);
          const func = Types.TypeRegistry.Get(kind);
          return func(schema2, value);
        }, (format, value) => {
          if (!Types.FormatRegistry.Has(format))
            return false;
          const func = Types.FormatRegistry.Get(format);
          return func(value);
        }, (value) => {
          return hash_1.ValueHash.Create(value);
```

**Comment by sinclairzx81** [comment]
@SaltyAom Hi!

> In an upcoming release of Elysia, Code Generation will be disabled by default on Cloudflare Worker, you can try out on Elysia 0.6.0-alpha.2 to see if it's solved the problem.

In Content-Security-Policy restricted environments (such as Cloudflare which omits the `unsafe-eval` directive), It is possible to swap out the `TypeCompiler` for the `Value` module (which implements dynamic type checking)

```typescript
import { Value } from '@sinclair/typebox/value'

const T  = Type.Object({ x: Number() })

const R = Value.Check(T, { x: 1 }) // no-eval

const E = Value.Errors(T, { x: '1' }).First() // no-eval
```
Which can be wrapped up with something like.
```typescript
const compile = <T extends TSchema>(schema: T, useJIT: boolean) => {
  if(useJIT) { // fast
    const check = TypeCompiler.Compile(schema)
    return (value: unknown) => { 
       if(check.Check(value)) return
       throw check.Errors(value).First() 
    }
  } else { // slow :(
     return (value: unknown) => { 
      if(Value.Check(schema, value)) return
      throw Value.Errors(schema, value).First() 
    }
  }
}
```
Unfortunately, using dynamic validation is going to much slower than JIT compiled schemas (by several orders), but will get you around the `eval` restriction on some environments. Alternatively, it might be possible to generate auxiliary modules using the compiler (by writing them to disk before importing them). This is some mix of AOT and JIT (and have been exploring the approach in a couple of projects), but mileage may vary for workers (which may not have access to disk to write), but just an idea if that's something worth exploring.

Anyway, hope this helps :)
S

**Comment by SaltyAom** [comment]
Setting aot to false in Elysia 0.6.0 stable should force typebox to use JIT instead of AoT, so this problem so be fixed.

See [Dynamic Mode](https://elysiajs.com/blog/elysia-06.html#dynamic-mode).

Please let me know if this solve your problem.

**Comment by implicit-invocation** [comment]
In my case, setting `aot` to `false` makes the errors go away.  
But `swagger` plugin will return `No operations defined in spec!`  
To make it work with cloudflare worker, I will have to figure out how to access `env` and type the `env` also (not only for variables but also for durable objects, d1, r2, KV).

**Comment by SaltyAom** [comment]
Hi, sorry I have just seen this update.

This should be fixed by updating Elysia 0.6.8

**Comment by implicit-invocation** [comment]
```
const app = new Elysia({ aot: false })
  .use(swagger())
  .derive((): { test: string } => ({
    test: "lorem ipsum",
  }))
  .get("/", (c) => {
    console.log(c.test);
    return "Hello Elysia";
  });
```
As posted on discord, swagger issue is fixed, but the code above will log `undefined`, seems that `{aot: false}` breaks `derive`.
